crates/wash-cli/src/cmd/dev/session.rs

```diff
@@ -432,3 +404,68 @@ impl WashDevSession {
         Ok((nats_child, wadm_child, wasmcloud_child))
     }
 }
+
+async fn get_host_id(log_path: PathBuf) -> anyhow::Result<String> {
+    let log_file = tokio::fs::File::open(&log_path)
+        .await
+        .with_context(|| format!("failed to open log file @ [{}]", &log_path.display()))?;
+
+    // looks for the two variations of the log line containing the host_id:
+    //   JSON: "host_id":"ABC123"
+    //   LOG:  host_id="ABC123"
+    let re = regex::Regex::new(r#"(?:\"host_id\":\s?\"|host_id=\")([A-Z0-9]+)\""#)
+        .context("failed to compile regex")?;
+
+    let mut lines = tokio::io::BufReader::new(log_file).lines();
+    loop {
+        if let Some(line) = lines
+            .next_line()
+            .await
+            .context("failed to read line from file")?
+        {
+            // if there's no captures, this line doesn't contain the host_id, keep looking
+            if let Some(captures) = re.captures(&line) {
+                return Ok(captures
+                    .get(1)
+                    .context("failed to get capture group")?
+                    .as_str()
+                    .to_string());
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::io::Write;
+    use tempfile::NamedTempFile;
+
+    #[tokio::test]
+    async fn test_get_host_id_from_standard_logging_pattern() {
+        let log_file = NamedTempFile::new().unwrap();
+        let log_path = log_file.path().to_path_buf();
+        let mut log_file = std::fs::OpenOptions::new()
+            .write(true)
+            .open(&log_path)
+            .unwrap();
+
+        writeln!(log_file, "2024-12-13T17:17:07.287574Z  INFO wasmcloud_host::wasmbus: wasmCloud host started host_id=\"ABC123\"").unwrap();
```

Comment:
When you're writing from an async context, you should use `tokio::fs` rather than the stuff in `std` since the tokio APIs are built to work with the event loop (a bit like using sync stuff from async JS).

I'm not sure we absolutely *need* async here, but if you do use it, you can probably boil this down to:

```rust
tokio::fs::write(&log_file, "2024-12-.....").await?;
assert_eq!(get_host_id(log_file).await? "ABC123");
```

Reply:
Also, for a follow up this might be a good thing to test with an integration test (usually those are the ones that attempt to do output to files and stuff) -- in general being able to start `wash dev` with the args that we're expecting should work identically to the [*existing* `wash dev` test](https://github.com/wasmCloud/wasmCloud/blob/main/crates/wash-cli/tests/wash_dev.rs#L46)